### DISCUSSION TITLE
Feature proposals

### DISCUSSION THREAD

**Discussion by tuur29** [notea-org/notea#90] [Ideas]
Hi there @QingWei-Li!

I recently discovered this awesome project! After looking at a lot of alternatives (wikijs, dokuwiki, outline to name a few) this seems like a perfect match for my needs. I have some minor improvements in mind that I'll try adding to a fork for personal use.
However I just wanted to run these by you first in case you would like me to create PR's for them in the main repo.

- Already created some dutch translations (#88)
- Currently working on adding a "editor width" setting (the [tailwind prose width](https://tailwindcss.com/docs/max-width) is a bit too limiting for me) (#91).
- Add directly link to image to view full resolution (#97)
- Changeable code syntax highlighting themes: Seems like a bug, also part of #97 for now
- Open item in tree button (#116)
- draw.io integration (#117)
- Use emojis in title as page icon (#118)
- Show path of a note (or part of) in search panel (#121)

My other ideas (in semi-order of importance):

- Support for running the app in a subpath (https://example.org/wiki for example)
- Table of contents view
- Allow deeplinks to search panel (integration with browser search)

Ideas that requires changes to rich-markdown-editor:

- Coloured status badges (to mark projects statuses, similar to the info/warning/tip blocks now)
- Collapsible panels
- Slack/Discord like emoji picker (using `:`) -> requires update for https://github.com/outline/rich-markdown-editor/pull/514

**Comment by HDGoldi** [comment]
The draw.io is an amazing idea! That would be very useful.

**Reply by tuur29** [reply]
Hi @HDGoldi! I wanted to let you know I made a small fix that allows you to embed published drawio links (https://github.com/QingWei-Li/notea/pull/117). It's by no means a full solution (links don't auto update after editing as far as I know), but it does allow you to preview your diagrams inside the note. I'm personally not a frequent user of drawio, so maybe it might be possible to save the data somewhere outside the url and generating an auto updating embed 🤔

Paste a link generated via `Publish` or `Save As > Link`
![image](https://user-images.githubusercontent.com/2472626/139581778-5ce785f0-db95-4db6-a57c-96a27bc1b660.png)

Then paste it and create an embed of the link:
https://user-images.githubusercontent.com/2472626/138601729-6b5eb07a-e79d-402d-bbcb-3756126d61a3.mp4
